packages/wagmi/src/react.tsx

```diff
@@ -33,5 +34,17 @@ export function useWeb3Modal() {
     await modal?.close()
   }
 
-  return { open, close }
+  function setThemeMode(themeMode: ThemeMode) {
```

Comment:
Let's create separate hook for theming i.e. `useWeb3ModalTheme`, this can export set functions for mode and variable changes.

Instead of
```ts
  const themeMode = modal?.getThemeMode()
  const themeVariables = modal?.getThemeVariables()
```

You should add these on client in addition to `getThemeMode` and `getThemeVariables` -> `subscribeThemeMode` and `subscribeThemeVariables`.

Then, create a state in `useWeb3ModalTheme` hook that sets up these subscribers and updates react / vue state. Don't forget to set unsubscribe as well to clean everything up when hooks unmount.